### PR TITLE
Support for DPS enrollment groups

### PR DESCRIPTION
**Pull request by Teodelas**
It's unclear how this sample works with DPS enrollment groups.  Added a comment to let users know that a device key must be generated from the DPS master key.  Note that other code samples take this into account and ask the user for a DPS Enrollment Group key and/or Individual enrollment key.

### CODE REVIEW

File: provisioning_client/samples/prov_dev_client_sample/prov_dev_client_sample.c [hunk 1]
```diff
@@ -103,6 +103,8 @@ int main()
     (void)prov_dev_security_init(hsm_type);
 
     // Set the symmetric key if using they auth type
+    //If using DPS with an enrollment group, this must the the derived key from the DPS Master key
```
Comment (ericwol-msft) [inline provisioning_client/samples/prov_dev_client_sample/prov_dev_client_sample.c]: derived device key

File: provisioning_client/samples/prov_dev_client_sample/prov_dev_client_sample.c [hunk 2]
```diff
@@ -103,6 +103,8 @@ int main()
...
+    //https://docs.microsoft.com/en-us/azure/iot-dps/how-to-legacy-device-symm-key#derive-a-device-key
```
Comment (CIPop) [inline provisioning_client/samples/prov_dev_client_sample/prov_dev_client_sample.c:107]: We could point to the updated documentation here instead: https://docs.microsoft.com/en-us/azure/iot-dps/concepts-symmetric-key-attestation?tabs=azure-cli#group-enrollments

**Comment by CIPop** [comment]
Can't address PR feedback here because the original branch got deleted. 
Ported everything to #2145. Closing this one.

**Pull request closed without merging**